documentation/bootstrap_upgrade_notes.md

```diff
@@ -0,0 +1,39 @@
+# Bootstrap upgrade notes
+
+**NOTE: If you have previously disabled your CA, please read this.**
+
+In the past, puppetserver users who have modified their `bootstrap.cfg` file to disable the CA on their compile masters and support a multi-master setup have been subject to a painful upgrade experience.
+
+Starting in puppetserver 2.5.0 this problem will start to be resolved. However, it's important to note that users will have to go through one final annoying upgrade if they're managing the CA service through their `bootstrap.cfg`. This page will walk you through what upgrading will be like if you are managing `bootstrap.cfg` yourself. 
+
+# Upgrading to 2.5.0 or newer
+If you are not managing your `bootstrap.cfg` file, you don't need to do anything.
+
+If you are managing your `bootstrap.cfg`, here's what you need to be aware of:
+
+## `ca.cfg`
+puppetserver will no longer look at `/etc/puppetlabs/puppetserver/bootstrap.cfg` for bootstrap entries. If you have modified `bootstrap.cfg` to disable the CA service, you will have to edit `/etc/puppetlabs/puppetserver/services.d/ca.cfg` and disable it there.
+
+It's a good idea to create `/etc/puppetlabs/puppetserver/services.d/ca.cfg` with the CA disabled *before* upgrading, since the puppetserver service will be restarted on upgrade if the service running when you upgrade. If you don't, the default `ca.cfg` created during the upgrade will re-enable the CA service.
```

Comment:
You could provide example contents of this file; current verbiage sort of assumes that users know exactly how to do that, and since the file will be short, seems like it'd be worth just giving a concrete example.